h2o-algos/src/main/java/hex/glm/ComputationState.java

```diff
@@ -459,7 +576,10 @@ public GradientInfo getGradient(double[] beta) {
 
   public void setBetaMultinomial(int c, double [] beta, double [] bc) {
     if(_u != null) Arrays.fill(_u,0);
-    fillSubRange(_activeData.fullN()+1,c,_activeDataMultinomial[c].activeCols(),bc,beta);
+    if (_parms._remove_collinear_columns)
+      fillSubRange(_betaLengthPerClass,c,_activeDataMultinomial[c].activeCols(),bc,beta);
+    else
+      fillSubRange(_activeData.fullN()+1,c,_activeDataMultinomial[c].activeCols(),bc,beta);
```

Comment:
Same here, could we have just `fillSubRange(_betaLengthPerClass,c,_activeDataMultinomial[c].activeCols(),bc,beta);`?

Reply:
Nope, _activeData.fullN() changes if inactive columns are removed.